### ISSUE TITLE
Implement deterministic semantic initialization from the Bithkuil semantic ABI

### ISSUE THREAD

**Issue by crates** [entif-ai/bithkuil#62]
## Summary

Implement a deterministic semantic-initialization path for the Bithkuil student so "trained from scratch" means **no inherited pretrained capability**, not "all semantic coordinates begin as unrelated random noise."

The integrated Bithkuil treatment should begin with token/embedding geometry compiled from the versioned semantic substrate. IID random initialization remains a required attribution/control condition.

## Contract

The primary initializer consumes only project-owned, versioned semantic inputs:

- #7 semantic ABI and canonical AST
- #8 grammar-derived semantic-factor map
- #9 cognitive algebra
- #11 token ABI
- the bounded accepted semantic graph produced by #53 when that graph is part of the selected experimental epoch

It MUST NOT import pretrained task-capable model weights, teacher hidden states, SBERT-style embeddings, or frontier-model embedding vectors into the primary student.

A useful abstract contract is:

```text
semantic object s
  -> deterministic semantic feature map phi(s)
  -> width-matched initialization transform
  -> initial embedding/state vector e(s)
```

The exact feature basis is versioned. Candidate dimensions include semantic family, subtype, role, arity, scope behavior, polarity, modality, evidential status, null/default/unknown distinctions, prerequisite ancestry, typed graph neighborhood, sibling/exclusion relations, and algebraic/operator properties.

The initializer may support a trainable residual after initialization. Any stochastic residual must be separately configured, seeded, and reported. The reference profile should remain deterministic.

## Required initialization profiles

Implement at least these named profiles behind one stable interface:

- `semantic_aligned`: semantic factors are mapped to their intended token identities.
- `iid_random`: ordinary seeded random initialization for attribution and compatibility controls.
- `semantic_permuted`: reuse the same compiled semantic-vector multiset but assign vectors to the wrong token identities through a deterministic bijection.

Later factor-ablated or graph-only profiles may be added after the first attribution result. Do not widen the first implementation merely to populate a large ablation catalog.

## Scratch-training definition

For this program:

> **Scratch-trained means the learner inherits no task-capable pretrained neural weights. It does not require the project to discard semantic structure known before gradient step one.**

The initialization manifest must make that distinction machine-readable.

## Geometry and reproducibility

Every initialization artifact must record:

- semantic ABI/version hashes
- token ABI hash
- initializer profile and version
- feature-map schema/version
- projection/transform algorithm and seed if any
- embedding width and parameter class
- output tensor hash
- reserved/padding-token treatment
- any residual-noise policy

The same inputs must reproduce the same initialized tensors within the declared numerical tolerance.

#42 will measure the resulting pre-training neural geometry. #28 owns causal attribution among aligned, random, permuted, and factor-access controls.

## Ternary compatibility

#19 must declare whether embeddings remain full precision or are quantized in a given precision arm. A ternary-forward path must not silently erase the semantic initialization before training begins.

If a later precision path quantizes initialized embeddings, it must measure the geometry distortion and treat that as an explicit intervention.

## Acceptance criteria

- [ ] One deterministic semantic initializer compiles the bounded Bithkuil semantic substrate into the student embedding width.
- [ ] The initializer does not consume pretrained task-capable model weights or pretrained embedding vectors.
- [ ] `semantic_aligned`, `iid_random`, and `semantic_permuted` are selectable through one initialization contract.
- [ ] Same manifest and seed reproduce initialization within the declared tolerance.
- [ ] Manifest hashes bind semantic ABI, token ABI, feature map, transform, and output tensors.
- [ ] Reserved and padding tokens have explicit initialization rules.
- [ ] Unit tests prove that semantic permutation preserves the vector multiset while breaking token-to-semantic alignment.
- [ ] #42 can inspect the initialized geometry before any optimizer step.
- [ ] #18 can instantiate the same student architecture under each initialization profile.
- [ ] #19 cannot silently change embedding precision or initialization semantics.
- [ ] Random initialization is documented as a control, not as the definition of scratch training.

## Dependencies

#7, #8, #9, #11. Consume the bounded #53 graph snapshot when the selected experimental epoch uses it.

## Downstream

#18 student baseline, #19 ternary compatibility, #22 SYS-01 plan lock, #23 G0, #28 attribution controls, #42 geometry audit, #52 local qualification.

Parent roadmap: #5.

## Workability

`EXECUTABLE_LEAF`: one bounded outcome, deterministic semantic initialization plus the minimum control-compatible interface needed by the existing student and attribution plan.
